src/test/fakeSuite/extension.test.ts

```diff
@@ -1,286 +1,330 @@
+            configuration.setPreConfigureScript(path.join(vscode.workspace.rootPath || "./", ".vscode/preconfigure_nonwin.sh"));
+            await make.preConfigure(make.TriggeredBy.tests);
+
+            configuration.prepareConfigurationsQuickPick();
+            configuration.setConfigurationByName(process.platform === "linux" ? "8cc_linux" : "8cc_mingw");
+            const retc: number = await make.cleanConfigure(make.TriggeredBy.tests, true);
+
+            configuration.setBuildBeforeLaunch(false);
+            const launchConfigurations: string[] = ["8cc()"];
+            for (const config of launchConfigurations) {
+                await configuration.setLaunchConfigurationByName(vscode.workspace.rootPath + ">" + config);
+                let status: string = await launch.getLauncher().validateLaunchConfiguration(make.Operations.debug);
+                let launchConfiguration: configuration.LaunchConfiguration | undefined;
+                if (status === launch.LaunchStatuses.success) {
+                    launchConfiguration = configuration.getCurrentLaunchConfiguration();
+                }
+
+                if (launchConfiguration) {
+                    launch.getLauncher().prepareDebugCurrentTarget(launchConfiguration);
+                    launch.getLauncher().prepareRunCurrentTarget();
+                }
+            }
+
+            configuration.setTargetByName("all");
+            make.prepareBuildTarget("all");
+
+            // Compare the output log with the baseline
+            // TODO: incorporate relevant diff snippets into the test log.
+            // Until then, print into base and diff files for easier viewing
+            // when the test fails.
+            let parsedPath: path.ParsedPath = path.parse(extensionLogPath);
+            let baselineLogPath: string = path.join(parsedPath.dir, process.platform === "linux" ? "../8cc_linux_baseline.out" : "../8cc_mingw_baseline.out");
+            let extensionLogContent: string = util.readFile(extensionLogPath) || "";
+            let baselineLogContent: string = util.readFile(baselineLogPath) || "";
+            let extensionRootPath: string = path.resolve(__dirname, "../../../../");
+            baselineLogContent = baselineLogContent.replace(/{REPO:VSCODE-MAKEFILE-TOOLS}/mg, extensionRootPath);
+            // fs.writeFileSync(path.join(parsedPath.dir, "base.out"), baselineLogContent);
+            // fs.writeFileSync(path.join(parsedPath.dir, "diff.out"), extensionLogContent);
+            expect(extensionLogContent).to.be.equal(baselineLogContent);
+        });
+    }
+
+    // dry-run logs for https://github.com/FidoProject/Fido.git
+    if (systemPlatform === "linux" || systemPlatform === "mingw") {
+        test(`Fido - ${systemPlatform}`, async() => {
+            // Settings reset from the previous test run.
+            extension.getState().reset(false);

```

Comment:
A lot of the linux test code looks like same between test cases.  Can you open an issue to look into refactoring some of this into a shared, parameterizable method?

Reply:
Yes, added a few improvement bullets in issue https://github.com/microsoft/vscode-makefile-tools/issues/193. After more triaging and discussion I can split that into more, especially if targeting different milestones.